iceoryx_dust/test/moduletests/test_cxx_string_conversion.cpp

```diff
@@ -108,4 +110,621 @@ TYPED_TEST(StdString_test, StringOfSizeCapaToSTDStringConvResultsInSizeCapa)
+
+    constexpr uint64_t TEST_CHAR_ARRAY_CAPACITY = STRINGCAP + 6U;
+    // NOLINTJUSTIFICATION required to verify string literal functionality of iox::string
+    // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
+    char testCharArray[TEST_CHAR_ARRAY_CAPACITY];
+    for (auto& c : testCharArray)
+    {
+        c = 'M';
+    }
+    testCharArray[STRINGCAP] = '\0';
+
+    const std::string testStdString = &testCharArray[0];
+    EXPECT_THAT(this->testSubject < testStdString, Eq(false));
+    EXPECT_THAT(this->testSubject <= testStdString, Eq(true));
+    EXPECT_THAT(this->testSubject > testStdString, Eq(false));
+    EXPECT_THAT(this->testSubject >= testStdString, Eq(true));
+
+    EXPECT_THAT(testStdString < this->testSubject, Eq(false));
+    EXPECT_THAT(testStdString <= this->testSubject, Eq(true));
+    EXPECT_THAT(testStdString > this->testSubject, Eq(false));
+    EXPECT_THAT(testStdString >= this->testSubject, Eq(true));
+}
+
+TYPED_TEST(StdString_test, AppendStdStringContainingNullWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f8814d78-449b-4c3a-b7c7-7c3ff2a0a62f");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    constexpr auto RESULT_CAPACITY = STRINGCAP + 10U;
+    const std::string expectedString{"ice\0ryx", 7};
+
+    string<RESULT_CAPACITY> sut("i");
+    const string<RESULT_CAPACITY> testCxxString(TruncateToCapacity, expectedString.substr(1).c_str(), 6U);
+    const std::string testStdString = expectedString.substr(1);
+
+    // append std::string
+    sut = "i";
+    sut.append(TruncateToCapacity, testStdString);
+    EXPECT_THAT(sut.capacity(), Eq(RESULT_CAPACITY));
+    EXPECT_THAT(sut.size(), Eq(7U));
+    EXPECT_THAT(std::memcmp(sut.c_str(), expectedString.c_str(), sut.size()), Eq(0));
+}
+
+TYPED_TEST(StdString_test, FindStdStringInEmptyStringFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "75e34d61-be16-4892-a931-96a96bc1e45f");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    string<STRINGCAP + 5U> testString("a");
+    auto res = this->testSubject.find(testString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    std::string testStdString = "a";
+    res = this->testSubject.find(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+/// @note template <typename T>
+/// iox::cxx::optional<uint64_t> find_first_of(const T& t, uint64_t pos = 0) const noexcept
+TYPED_TEST(StdString_test, FindFirstOfFailsForEmptyStdStringInEmptyString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "207671e4-cef3-40d2-8984-e8ae5c2b42ec");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    string<STRINGCAP + 5U> testString;
+    auto res = this->testSubject.find_first_of(testString);
+    EXPECT_THAT(res.has_value(), Eq(false));
```

Comment:
This is not needed here.

Reply:
True, this is already in `FindFirstOfFailsForEmptyStringInEmptyString` with test id `21f90f13-6b15-4ce1-9258-c21154b6043c`.